solr/core/src/java/org/apache/solr/handler/component/CloudReplicaSource.java

```diff
@@ -170,10 +170,9 @@ private List<String> findReplicas(
               .filter(replica -> replica.isActive(clusterState.getLiveNodes()))
               .filter(
                   replica ->
-                      !builder.onlyNrt
-                          || (replica.getType() == Replica.Type.NRT
-                              || (replica.getType() == Replica.Type.TLOG
-                                  && isShardLeader.test(replica))))
+                      !builder.onlyRealTime
+                          || (replica.getType().follower && !replica.getType().followerSkipCommit)
```

Comment:
This is statement is (in my opinion) harder to understand than the code it is replacing.
A replica is considered real time if it can be a follower but the follower doesn't skip commit (whatever that means)? I don't get it.

In any case, the logic telling if a replica type is real time or not based on replica type attributes likely belongs in the `Replica.Type` enum, like the other attributes.